envi/config.py

```diff
@@ -80,6 +80,48 @@ def getOptionDoc(self, optname):
         '''
         return self.cfgdocs.get(optname)
 
+    def getConfigPaths(self, with_vals=False):
+        '''
+        Return a list of valid path strings
+        If "with_vals" is True, also include the current values
+        '''
+        paths = []
+        todo = [ ([], self) ]
+
+        while todo:
+            path, config = todo.pop()
+            
+            if config.keys():
+                pathstr = '.'.join(path) + "."
+                if with_vals:
+                    newpaths = [ "%s%s = %s" % (pathstr, key, config[key])  for key in config.keys()]
```

Comment:
that was my first thought after reading the last comment.  i was also thinking about a "-O ?" option that just spits it out.